### PR TITLE
chore(release): v1.1.2

### PR DESCRIPTION
**Pull request by avmturo**
### Bug Fixes

* [ORI-1381] fix uid return type to {uid: string}. Fix return type of editAsset (only returns success status) ([30fbf65](https://github.com/getoriginal/original-js/commit/30fbf65c07d1e32ef5a4960aa482f3eccbbdbf56))
* cleanup return types on get...ByUserId, minor cleanups on method doc strings ([a335dcc](https://github.com/getoriginal/original-js/commit/a335dcca8308e0a675295a6154158aec1348a4cb))
* get rid of ApiSearchResponse type and use ApiResponse type throughout class. ([91f57e4](https://github.com/getoriginal/original-js/commit/91f57e4794d7cc96284ef956cdc2be3163e444d1))
* README add link to original docs ([45c0b16](https://github.com/getoriginal/original-js/commit/45c0b163df2c392e73d97efa230904e29293c69e))
* README updates ([a6e889b](https://github.com/getoriginal/original-js/commit/a6e889b87c80f4b27731a7b67ea39206ff3bc191))
* refactor getAssetsByUserId to getAssetsByUserUid to be consistent with naming conventions. ([0808700](https://github.com/getoriginal/original-js/commit/0808700be21f55ffc84f223280008b13a7d462d1))

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -254,10 +254,10 @@ export class Original {
    * editAsset
    * @param {string} uid uid of the asset to edit
    * @param {EditAssetParams} asset The details of the asset to be edited
-   * @return {Promise<APIResponse<Uid>>} Returns the uid of the edited asset
+   * @return {Promise<APIResponse<uidResponse>>} Returns the uid of the edited asset
    */
   async editAsset(uid: string, asset: EditAssetParams) {
-    return await this._put<APIResponse<Uid>>(`asset/${uid}`, asset);
+    return await this._put<APIResponse<uidResponse>>(`asset/${uid}`, asset);
```
Comment (don1989) [inline src/client.ts:260]: Edit does not return anything: https://docs.getoriginal.com/docs/edit-asset

File: src/client.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ import {
   TransferParams,
   UserParams,
   OriginalOptions,
-  Uid,
+  uidResponse,
```
Comment (don1989) [inline src/client.ts:13]: Lets capitalise all types so its consistent everywhere

**Comment by don1989** [comment]
Why do we have the lint failing btw?

File: README.md
Comment (don1989) [inline README.md:1]: Typo in this location:

// gets assets the owner uid and a

Comment (don1989) [reply README.md:1]: We also need to add Edit to the readme

Comment (don1989) [reply README.md:1]: Actually I see we have edit with the Asset endpoints. nvm.

**Review by don1989 — APPROVED** [review]
lgtm once the last typo is fixed

File: src/client.ts [hunk 3]
```diff
@@ -243,21 +243,21 @@ export class Original {
...
+  async getAssetsByUserUid(userUid: string) {
+    return await this._get<APISearchResponse<Asset[]>>('asset', { user_uid: userUid });
   }
 
   /**
    * editAsset
    * @param {string} uid uid of the asset to edit
    * @param {EditAssetParams} asset The details of the asset to be edited
-   * @return {Promise<APIResponse<Uid>>} Returns the uid of the edited asset
+   * @return {Promise<{success: boolean}>} Returns success status of the edit
```
Comment (don1989) [inline src/client.ts:257]: This should be `APIResponse<null>` https://docs.getoriginal.com/docs/edit-asset

```
{
    "success": true,
    "data": null
}
```

Comment (don1989) [reply src/client.ts:257]: Btw, do we need APISearchResponse ? why not add `| null` to the data on APIResponse?

Comment (don1989) [reply src/client.ts:257]: You could also leave APIResponse as is I believe, and just pass `null` as the template arg

File: test/sdk/e2e.js [hunk 4]
```diff
@@ -79,7 +79,7 @@ describe('Original sdk e2e-method tests', async () => {
 
 	it('gets assets by user uid', async () => {
 		const original = new Original(apiKey, apiSecret, { baseURL: 'https://api-acceptance.getoriginal.com/api/v1' });
```
Comment (don1989) [inline test/sdk/e2e.js:81]: baseURL: should use an env var

File: src/client.ts [hunk 5]
```diff
@@ -183,22 +182,22 @@ export class Original {
    * getUserByEmail
    *
    * @param {String} email Email of the user to get
-   * @return {Promise<APISearchResponse<User>>} Returns the details of the user,
+   * @return {Promise<APIResponse<User | null>>} Returns the details of the user,
```
[the remark this answers is not in the document]
Comment (don1989) [reply src/client.ts:185]: Or an empty array?

[the remark this answers is not in the document]
Comment (avmturo) [reply src/client.ts:185]: We have a test covering this. But it is a good catch, for the getUserBy... we return { data: null }
but
for assets/transfer/burn getBy... we retun an empty array. I just added a test for this.

**Comment by avmturo** [comment]
ReadMe doesn't have a link to the actual documentation, should add it here.

**Pull request merged**